include/picongpu/param/density.param

```diff
@@ -223,24 +223,32 @@ namespace picongpu
         /* definition of sphere profile with flanks */
         using SphereFlanks = SphereFlanksImpl<SphereFlanksParam>;
 
-
+        /** Density values taken from a file
+         *
+         * The file follows the openPMD standard.
+         * The density values are written as a dataset of type float_X.
+         * Each value defines density in the cell with the corresponding total coordinate minus the given offset.
+         * When the functor is instantiated, it will load the part matching the current domain position.
+         * If the file dataset is not large enough, it will be padded with 0.
+         * Dimensionality of the file indexing must match the simulation dimensionality.
+         * Density values are in BASE_DENSITY_SI units.
+         */
         PMACC_STRUCT(
-            FromHDF5Param,
-            /* prefix of filename
-             * full file name: gas_0.h5
-             * filename = "gas"
-             * iteration = 0
-             */
-            (PMACC_C_STRING(filename, "gas"))
+            FromOpenPMDParam,
+            //! Name for the openPMD file
+            (PMACC_C_STRING(filename, "density.h5"))
 
-                (PMACC_C_STRING(datasetName, "fields/e_chargeDensity"))
+            //! Name of the dataset inside the file
+            (PMACC_C_STRING(datasetName, "density"))
 
-            /* simulation step*/
-            (PMACC_C_VALUE(uint32_t, iteration, 0))(
-                PMACC_C_VALUE(float_X, defaultDensity, 0.0))); /* struct FromHDF5Param */
+            //! Iteration inside the file (only file, not related to the current simulation time iteration)
+            (PMACC_C_VALUE(uint32_t, iteration, 0))
 
-        /* definition of cloud profile */
-        using FromHDF5 = FromHDF5Impl<FromHDF5Param>;
+            //! Offset in the file in cells: each value is density at (total cell index - offset)
+            (PMACC_C_VECTOR_DIM(int, simDim, offset, 0, 0, 0))); /* struct FromOpenPMDParam */
```

Comment:
I am not sure how to write that in a way that works for 2d and 3d

Reply:
This is automatically working for 2D and 3D because it is using `simDim`. In the code this objects looks like a `Vector<int,simDim>`.

Reply:
Yes, but how there are three values `0`.

Reply:
IN case `simDim` is `2` the last zero is ignored.